sklearn/compose/_column_transformer.py

```diff
@@ -496,6 +515,8 @@ def fit_transform(self, X, y=None):
         # TODO: this should be `feature_names_in_` when we start having it
         if hasattr(X, "columns"):
             self._feature_names_in = np.asarray(X.columns)
+            self._only_str_columns = all(isinstance(cols, str)
+                                         for cols in self._feature_names_in)
```

Comment:
```suggestion
            self._only_str_columns = all(isinstance(col, str)
                                         for col in self._feature_names_in)
```
Is there a simpler test based on the dtype  of `self._feature_names_in` to check that it is a string?

Reply:
numpy has a native string type that can be use to quickly check that the ndarray is a string:

```py
import numpy as np

x = np.array(['a', 'b'])
x
array(['a', 'b'], dtype='<U1')
```

Although, usually when converting a dataframe's column into an ndarray, the dtype is an object:

```py
import pandas as pd
import numpy as np

df = pd.DataFrame({'a': [1, 2,3], "b": [3, 4, 5]})

np.asarray(df.columns)
# array(['a', 'b'], dtype=object)
```

This would require iterating over all the elements to check if it is a string.

Reply:
💡